### PR TITLE
Attempt to fix crash on ProductListToolbarHelper

### LINKED ISSUE #11089: IllegalStateException: Fragment NavHostFragment{f34d844} (94f9f909-e3a2-43f2-850e-49f4e6ec429d) has not been attached yet.
Sentry Issue: [WOOCOMMERCE-ANDROID-9NT](https://a8c.sentry.io/issues/5065645661/?referrer=github_integration)

```
IllegalStateException: Fragment NavHostFragment{f34d844} (94f9f909-e3a2-43f2-850e-49f4e6ec429d) has not been attached yet.
    at com.woocommerce.android.ui.main.MainActivity.getHostChildFragment(MainActivity.kt:489)
    at com.woocommerce.android.ui.main.MainActivity.isAtTopLevelNavigation(MainActivity.kt:594)
    at com.woocommerce.android.ui.main.MainActivity.isChildFragmentShowing(MainActivity.kt:458)
    at com.woocommerce.android.ui.products.ProductListToolbarHelper.shouldShowSearchMenuItem(ProductListToolbarHelper.kt:200)
    at com.woocommerce.android.ui.products.ProductListToolbarHelper.refreshOptionsMenu(ProductListToolbarHelper.kt:167)
...
(12 additional frame(s) were not displayed)
```

### LINKED ISSUE #11480: IllegalStateException: Fragment NavHostFragment{883b129} (46cff297-797a-496e-a1ad-fced1dc46ebf) has not been attached yet.
Sentry Issue: [WOOCOMMERCE-ANDROID-APH](https://a8c.sentry.io/issues/5316250541/?referrer=github_integration)

```
IllegalStateException: Fragment NavHostFragment{883b129} (46cff297-797a-496e-a1ad-fced1dc46ebf) has not been attached yet.
    at com.woocommerce.android.ui.main.MainActivity.getHostChildFragment(MainActivity.kt:493)
    if (navHostFragment?.childFragmentManager?.fragments?.isNotEmpty() == true) {
    at com.woocommerce.android.ui.main.MainActivity.isAtTopLevelNavigation(MainActivity.kt:598)
    val activeChild = getHostChildFragment()
    at com.woocommerce.android.ui.main.MainActivity.isChildFragmentShowing(MainActivity.kt:462)
    !isAtTopLevelNavigation(isAtRoot = isAtNavigationRoot(), destination = it)
    at com.woocommerce.android.ui.products.list.ProductListToolbarHelper.shouldShowSearchMenuItem(ProductListToolbarHelper.kt:211)
    val isChildShowing = (activity as? MainNavigationRouter)?.isChildFragmentShowing() ?: false
    at com.woocommerce.android.ui.products.list.ProductListToolbarHelper.refreshOptionsMenu(ProductListToolbarHelper.kt:178)
    val showSearch = shouldShowSearchMenuItem()
...
(11 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by hichamboushaba**
Closes: #11089 Closes: #11480

### Description
This is just an attempt to fix the crash occurring on `ProductListToolbarHelper`, my theory of the crash is this: "if we check the events in Sentry, we can see that it always occur during a configuration change, and since we `post` the action to refresh the menu to the `Toolbar`'s handler, there seems to be a race condition where the posting happens, and then a configuration change happens before the callback is executed, which means that when it does, the app's `NavHostFragment` could be still not attached"

To solve this, I did two changes to the `ProductListToolbarHelper`:
1. I updated it to observe the View's `Lifecycle` instead of the Fragment's one, this is needed to ensure `onDestroy` is called during the View's detachment, and not after.
2. I added logic to remove the menu refresh Runnable from the `Toolbar`'s Handler when the view is destroyed.

### Testing instructions
I couldn't reproduce the crash, so it's just code review, and ensuring nothing breaks.

### Images/gif

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by hichamboushaba** [comment]
@kidinov @jd-alexander I requested your review since you seem to have some past background about this, I'm keeping the PR as draft for now, my goal is just to start a discussion, and see if the theory above makes sense, then discuss if the fix is the appropriate one.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/list/ProductListToolbarHelper.kt [hunk 1]
```diff
@@ -86,7 +90,7 @@ class ProductListToolbarHelper @Inject constructor(
         this.viewModel = productListViewModel
         this.binding = binding
 
-        fragment.lifecycle.addObserver(this)
+        fragment.viewLifecycleOwner.lifecycle.addObserver(this)
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/list/ProductListToolbarHelper.kt:93]: This change also ensures that we don't leak the fragment's view, we were setting `binding` to `null` in the fragment's `onDestroy` due to tying it to its `Lifecycle`, but now it'll be executed correctly on a matching callback of `onDestroyView`.

**Review by kidinov — APPROVED** [review]
@hichamboushaba Thank you! The potential fix makes sense to me

Have you considered adding a line in the release notes?

**Comment by hichamboushaba** [comment]
Thanks @kidinov for the review.

> Have you considered adding a line in the release notes?

Done.

**Pull request merged**